frontend/src/concepts/pipelines/content/pipelineSelector/utils.ts

```diff
@@ -0,0 +1,86 @@
+import { SearchInputProps } from '@patternfly/react-core';
+import * as React from 'react';
+import { GetColumnSort, SortableData } from '~/components/table';
+import usePipelineFilter, {
+  FilterOptions,
+  getDataValue,
+} from '~/concepts/pipelines/content/tables/usePipelineFilter';
+import { PipelinesFilter } from '~/concepts/pipelines/types';
+
+type SelectorTableColumnSortProps<DataType> = {
+  columns: SortableData<DataType>[];
+  sortField?: string;
+  setSortField: (field: string) => void;
+  sortDirection?: 'asc' | 'desc';
+  setSortDirection: (dir: 'asc' | 'desc') => void;
+};
+
+export const getSelectorTableColumnSort =
+  <T>({
+    columns,
+    sortField,
+    sortDirection,
+    setSortField,
+    setSortDirection,
+  }: SelectorTableColumnSortProps<T>): GetColumnSort =>
+  (columnIndex: number) =>
+    columns[columnIndex].sortable
+      ? {
+          sortBy: {
+            index: columns.findIndex((c) => c.field === sortField),
+            direction: sortDirection,
+            defaultDirection: 'asc',
+          },
+          onSort: (_event, index, direction) => {
+            setSortField(String(columns[index].field));
+            setSortDirection(direction);
+          },
+          columnIndex,
+        }
+      : undefined;
+
+type SelectorSearchProps = {
+  setFilter: (filter?: PipelinesFilter) => void;
+  fetchedSize: number;
+  loaded: boolean;
+};
+
+export const useSelectorSearch = ({
+  setFilter,
+  fetchedSize,
+  loaded,
+}: SelectorSearchProps): Required<Pick<SearchInputProps, 'onChange' | 'onClear'>> & {
+  value?: string;
+  totalSize: number;
+} => {
+  const totalSizeRef = React.useRef(0);
+  const { onFilterUpdate, filterData, onClearFilters } = usePipelineFilter(setFilter);
+
+  const search = getDataValue(filterData[FilterOptions.NAME]);
+
+  const totalSize = React.useMemo(() => {
+    if (loaded && !search) {
+      totalSizeRef.current = fetchedSize;
+      return fetchedSize;
+    }
+    return totalSizeRef.current;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loaded]);
```

Comment:
Please always add comments when disabling an eslint rule. In particular this memo solves a specific use case where we need to catch the total size.